### PR TITLE
Deinitialize in onDetachedFromEngine on Android

### PR DESCRIPTION
**Pull request by Rexios80**
Without this, devices stay connected to detached native code and behave unexpectedly when reattached to flutter

### CODE REVIEW

File: packages/reactive_ble_mobile/android/src/main/kotlin/com/signify/hue/flutterreactiveble/PluginController.kt [hunk 1]
```diff
@@ -71,6 +71,11 @@ class PluginController {
         bleStatusChannel.setStreamHandler(bleStatusHandler)
     }
 
+    internal fun deinitialize() {
```
Comment (Taym95) [inline packages/reactive_ble_mobile/android/src/main/kotlin/com/signify/hue/flutterreactiveble/PluginController.kt:74]: can be private

**Review by Taym95 — CHANGES_REQUESTED** [review]

**Comment by Rexios80** [comment]
This crashes if there aren't any devices connected

**Comment by Rexios80** [comment]
Fixed that. Lateinit variables are dangerous.

Also I used the current instance of the plugin as the method handler. For this plugin it doesn't look like that changes anything, but it matches the new flutter plugin template and can prevent issues in the future.

**Review by remonh87 — APPROVED** [review]
yes much better thanks!!

**Review by Taym95 — APPROVED** [review]

**Comment by Rexios80** [comment]
Any idea when this will be released?

**Comment by Taym95** [comment]
@Rexios80 we will release a new version ASAP

**Pull request merged**